File: solution-service/src/main/kotlin/codemaster/servicies/solution/domain/model/Difficulty.kt
package codemaster.servicies.solution.domain.model

sealed class Difficulty(val name: String) {
    data object Easy : Difficulty("easy")
    data object Medium : Difficulty("medium")
    data object Hard : Difficulty("hard")

    companion object {
        fun from(name: String): Difficulty = when (name.lowercase()) {
            "easy" -> Easy
            "medium" -> Medium
            "hard" -> Hard
            else -> throw IllegalArgumentException("Unknown difficulty: $name")
        }
    }
}

File: codequest-service/src/main/nodejs/codemaster/servicies/codequest/domain/codequest/difficulty.ts
export class Difficulty {
  private constructor(public readonly name: string) {}

  static readonly Easy = new Difficulty("Easy");
  static readonly Medium = new Difficulty("Medium");
  static readonly Hard = new Difficulty("Hard");

  static from(value: string): Difficulty {
    switch (value.toLowerCase()) {
      case "easy": return Difficulty.Easy;
      case "medium": return Difficulty.Medium;
      case "hard": return Difficulty.Hard;
      default:
        throw new Error(`Invalid difficulty: ${value}`);
    }
  }

  equals(other: Difficulty): boolean {
    return this.name === other.name;
  }
}
